### ISSUE TITLE
Question: Compatible version of Corundum

### ISSUE THREAD

**Issue by nightseas** [liuw666-bruce/DPDK-with-Corundum#2]
Hi,

Just would like to know which version/commit of Corundum was tested with this repo?

Thanks & BR.

**Comment by liuw666-bruce** [comment]
I downloaded the project of corundum at 2020.11.2

**Comment by nightseas** [comment]
Hi,

> The 2020.11 version of Corundum is actually an AXI lib instead of full functional NIC.
---

Updated, made a mistake on the commit HEAD. I'll try earlier version of FPGA and see if ti's still the same problem.

I tried with HEAD commit 38f7666, which can be successfully initialized. But Corundum only had TX traffic and no packet was received during pktgen test.

Any idea why the RX didn't work?
Kernel message show a question on interrupt. Does it mean that there's incorrect IRQ configuration on corundum?
```sh
$ dmesg
...
[ 2086.790533] uio_pci_generic 0000:17:00.0: No IRQ assigned to device: no support for interrupts?
[ 2106.338614] ice 0000:ca:00.0: Removed PTP clock
```

pktgen command:
```sh
sudo ./build/app/pktgen -v -c 0xFFF --proc-type auto --log-level=8 -- -P -v -m "[4/6:8/10].0,[5/7:9/11].1"
```

Hardware setup is CPU socket 0 <-> Port 0: Corundum (ADM-PCIe-9V3) <-> 100G DAC <-> Port 1: Intel E810 <-> CPU socket 1.
![image](https://user-images.githubusercontent.com/4987044/233764413-dace1508-f372-4602-ad38-4f2a4d0e5f65.png)

Thanks & BR.

**Comment by nightseas** [comment]
Same result with 2022 Oct version.

```sh
commit 53f4275ea2325df81d481782659b42dff0f9a3bd (HEAD)
Author: Alex Forencich <alex@alexforencich.com>
Date:   Tue Oct 13 23:52:52 2020 -0700

    Add output registers for I2C interface to improve timing
```

**Comment by liuw666-bruce** [comment]
> 

Did you try ARP protocol? The board works well with original Linux driver, right? And I don't think it is related to interrupt, this DPDK driver doesn't support interrupt.

**Comment by nightseas** [comment]
> > 
> 
> Did you try ARP protocol? The board works well with original Linux driver, right? And I don't think it is related to interrupt, this DPDK driver doesn't support interrupt.

Yes, only ARP can be received. And with kernel driver the board works fine.

**Comment by nightseas** [comment]
It's an issue of improper handling RSS in the driver. The RSS mask shall be dynamically configured based on how many cores/queues are used for RX.

**Comment by zqchinaone** [comment]
@nightseas which version/commit of Corundum was tested with this repo?  thanks

**Comment by nightseas** [comment]
[38f7666](https://github.com/nightseas/corundum/commit/38f766646b15c694c47de3c94bd416cb4e76156d) is the commit that I've been testing. It's just a fork from Corundum so that you will find the same commit on upstream repo.

I'm planning to port it to a later version of Corundum (before QP is introduced), and perhaps DPDK too.

**Comment by zqchinaone** [comment]
好的，多谢。

**Comment by liuw666-bruce** [comment]
> 

I think you are right, I set the rx_queue_count to 16 because the driver failed to alloc memory (too much) when the rx_queue_count is 8192. I didn't have testing devices to fix the issue now. Anybody who need to temporarily pass it can try to set the (especially for rx and rx complete)queue number to 16 in HW.

#define MQNIC_MAX_EVENT_RINGS   2//256
#define MQNIC_MAX_TX_RINGS      16//8192
#define MQNIC_MAX_TX_CPL_RINGS  16//8192
#define MQNIC_MAX_RX_RINGS     16// 8192
#define MQNIC_MAX_RX_CPL_RINGS  16//8192

**Comment by liuw666-bruce** [comment]
> It's an issue of improper handling RSS in the driver. The RSS mask shall be dynamically configured based on how many cores/queues are used for RX.

I think you are right, I set the rx_queue_count to 16 because the driver failed to alloc memory (too much) when the rx_queue_count is 8192. I didn't have testing devices to fix the issue now. Anybody who need to temporarily pass it can try to set the (especially for rx and rx complete)queue number to 16 in HW.

#define MQNIC_MAX_EVENT_RINGS 2//256
#define MQNIC_MAX_TX_RINGS 16//8192
#define MQNIC_MAX_TX_CPL_RINGS 16//8192
#define MQNIC_MAX_RX_RINGS 16// 8192
#define MQNIC_MAX_RX_CPL_RINGS 16//8192

**Comment by liuw666-bruce** [comment]
> > It's an issue of improper handling RSS in the driver. The RSS mask shall be dynamically configured based on how many cores/queues are used for RX.
> 
> I think you are right, I set the rx_queue_count to 16 because the driver failed to alloc memory (too much) when the rx_queue_count is 8192. I didn't have testing devices to fix the issue now. Anybody who need to temporarily pass it can try to set the (especially for rx and rx complete)queue number to 16 in HW.
> 
> #define MQNIC_MAX_EVENT_RINGS 2//256 #define MQNIC_MAX_TX_RINGS 16//8192 #define MQNIC_MAX_TX_CPL_RINGS 16//8192 #define MQNIC_MAX_RX_RINGS 16// 8192 #define MQNIC_MAX_RX_CPL_RINGS 16//8192

It is also OK to open more queues in HW, but don't forget to modify this MICROs in SW at the same time.
